index.js

```diff
@@ -36,7 +38,10 @@ function DNS (opts) {
   }
 
   function onlistening () {
-    self._interval = setInterval(ontimeout, 250)
+    var timeSlices = self._triesArray.reduce(function (prev, curr) {
+      return prev + curr
+    }, 0)
```

Comment:
could you move the reduce function outside the onlistening one to a generic 'add' function?